library/core/src/intrinsics.rs

```diff
@@ -2119,18 +2185,20 @@ extern "rust-intrinsic" {
 /// the occasional mistake, and this check should help them figure things out.
 #[allow_internal_unstable(const_eval_select)] // permit this to be called in stably-const fn
 macro_rules! assert_unsafe_precondition {
-    ($e:expr) => {
+    ($([$($tt:tt)*])?($($i:ident:$ty:ty),*$(,)?) => $e:expr) => {
         if cfg!(debug_assertions) {
-            // Use a closure so that we can capture arbitrary expressions from the invocation
-            let runtime = || {
+            // allow non_snake_case to allow capturing const generics
+            #[allow(non_snake_case)]
+            fn runtime$(<$($tt)*>)?($($i:$ty),*) {
```

Comment:
Maybe try adding `inline(always)` here?

Reply:
Also shouldn't this be `track_caller` to actually fix https://github.com/rust-lang/rust/issues/100696?

Reply:
Yeah, added.

Reply:
But this means the PR doesn't have a test for https://github.com/rust-lang/rust/issues/100696. Please always add a test when closing an issue.

Reply:
It does: https://github.com/rust-lang/rust/pull/100759/files#diff-14c656686badef9cc39ce6cf394aa18a0f61ccfe3d8c9e1c53b8cb9f1e791c57

It is just that that issue isn't caused by this one not having track_caller.